vcd/resource_vcd_snat.go

```diff
@@ -164,16 +187,77 @@ func resourceVcdSNATDelete(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
 	}
 
-	task, err := edgeGateway.RemoveNATMapping("SNAT", d.Get("internal_ip").(string),
-		d.Get("external_ip").(string),
-		"")
+	if "" != d.Get("network_name").(string) {
+		err = edgeGateway.RemoveNATRule(d.Id())
+		if err != nil {
+			return fmt.Errorf("error deleting SNAT rule: %#v", err)
+		}
+	} else {
+		// this for back compatibility when network name and network type isn't provided - TODO remove with major release
+		task, err := edgeGateway.RemoveNATMapping("SNAT", d.Get("internal_ip").(string),
+			d.Get("external_ip").(string),
+			"")
+		if err != nil {
+			return fmt.Errorf("error deleting SNAT rule: %#v", err)
+		}
+		err = task.WaitTaskCompletion()
+		if err != nil {
+			return err
+		}
+
+	}
+	return nil
+}
+
+func resourceVcdSNATUpdate(d *schema.ResourceData, meta interface{}) error {
+
+	vcdClient := meta.(*VCDClient)
+
+	// Update supports only when network name and network type provided
+	networkName := d.Get("network_name")
+	if nil == networkName || networkName.(string) == "" {
+		return fmt.Errorf("update works only when network_name and network_type is provided and rule created using them \n")
+	}
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
 	if err != nil {
-		return fmt.Errorf("error setting SNAT rules: %#v", err)
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
 	}
-	err = task.WaitTaskCompletion()
+
+	natRule, err := edgeGateway.FetchNatRule(d.Id())
 	if err != nil {
+		log.Printf("Error: Nat rule isn't found")
+		d.SetId("")
 		return err
```

Comment:
Would like to see more context rule ID

Reply:
It's a little complicated. For user it will mislead. Cause ind standard edge gtw you won't see Id's in advanced edge gtw it won't match. vCD uses <tags> element to map internal mapping between NSX rule mapping. So it will mislead exposing ID cause user won't find anywhere to match it in UI.

Reply:
Would it match what is in Terraform state file? If yes, it would still be useful to print.

Reply:
Yes, it would match state file ID

Reply:
Then I'd print it. If you think it may be misleading, maybe say it's "internal NSX ID" or something like that.

Reply:
updated